core/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -182,13 +183,18 @@ public final T read(Path file) throws IOException {
             try (final IndexInput indexInput = dir.openInput(file.getFileName().toString(), IOContext.DEFAULT)) {
                  // We checksum the entire file before we even go and parse it. If it's corrupted we barf right here.
                 CodecUtil.checksumEntireFile(indexInput);
-                CodecUtil.checkHeader(indexInput, STATE_FILE_CODEC, STATE_FILE_VERSION, STATE_FILE_VERSION);
+                final int fileVersion = CodecUtil.checkHeader(indexInput, STATE_FILE_CODEC, MIN_COMPATIBLE_STATE_FILE_VERSION,
+                    STATE_FILE_VERSION);
                 final XContentType xContentType = XContentType.values()[indexInput.readInt()];
-                indexInput.readLong(); // version currently unused
+                if (fileVersion == 0) {
```

Comment:
magic constant. would be useful to know which ES versions used this constant, see https://github.com/elastic/elasticsearch/pull/17987/files#diff-35f0f7b57ab6d12add07c462b761253bR67